samples/matter/lock/src/combine_ota_images.py

```diff
@@ -0,0 +1,158 @@
+            return multi.parse_header(tempImgFile)
+
+def extract_core_images(fullImage: bytes) -> list:
+    """
+    Extract app core and net core raw images from given NRF multi image binary representation
+    """
+    cborHeader = parse_multi_image_header(fullImage)
+    APP_CORE_ID = 0
+    offset = cborHeader["img"][APP_CORE_ID]['size']
+    bytesHeader = cbor2.dumps(cborHeader)
+    imageStart = len(bytesHeader) + len(struct.pack('<H', len(bytesHeader))) # the header is preceded by its size (encoded on 2 bytes)
+    headerLessImage = fullImage[imageStart:]
+    return headerLessImage[:offset], headerLessImage[offset:]
+
+def tlv_headers_equal(first: dict, second: dict) -> bool:
+    """
+    Verify if the relevant TLV header fields are the same across two provided JSONs
+    """
+    keys_to_compare = [ota.HeaderTag.VENDOR_ID, ota.HeaderTag.PRODUCT_ID,
+                       ota.HeaderTag.VERSION, ota.HeaderTag.VERSION_STRING]
+    return all(first.get(key) == second.get(key) for key in keys_to_compare)
+
+def generate_multi_image(images : list) -> bytes:
+    """
+    Generate the NRF multi image bytes based on provided image list
+    """
+    imageFiles = []
+    with tempfile.TemporaryDirectory() as tempDir:
+        for idx, image in enumerate(images):
+            outputFile = os.path.join(tempDir, f'outputFile{idx}')
+            imageFiles.append([idx, outputFile])
+            with open(outputFile, 'wb') as tempImgFile:
+                tempImgFile.write(image)
+        outputPath = os.path.join(tempDir, "multiImageFile")
+        multi.generate_image(imageFiles, outputPath)
+        with open(outputPath, 'rb') as result:
+            return result.read()
+
+def generate_ota_multi_variant_image(tlv: dict, multiImage: bytes, outputFile: str) -> None:
+    """
+    Generate the Matter OTA file based on the provided TLV header and NRF multi image bytes
+    """
+    with tempfile.TemporaryDirectory() as tempDir:
+        inputFile = os.path.join(tempDir, 'inputFile')
+        with open(inputFile, 'wb') as tempImgFile:
+            tempImgFile.write(multiImage)
+        otaArgs = GenerateOtaImageArgs(tlv, inputFile, outputFile)
+        ota.validate_header_attributes(otaArgs)
+        ota.generate_image(otaArgs)
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Matter OTA multi-variant application image utility')
+    parser.add_argument('input_file1', help='Path to the Thread image file')
+    parser.add_argument('input_file2', help='Path to the Wi-Fi image file')
+    parser.add_argument('output_file', help='Path to the output image file')
+
+    parseArgs = parser.parse_args()
+
+    _, _, threadHeaderSize, threadHeaderTlv = ota.parse_header(ParseHeaderArgs(parseArgs.input_file1))
+    _, _, wifiHeaderSize, wifiHeaderTlv  = ota.parse_header(ParseHeaderArgs(parseArgs.input_file2))
+
+    if not threadHeaderSize == wifiHeaderSize or not tlv_headers_equal(threadHeaderTlv, wifiHeaderTlv):
```

Comment:
I think we cannot require equal sizes. In some corner cases `payload_size` could be very different for both variants leading to different TLV encoding lengths.

Reply:
Hmm ok, that's right.